packages/@expo/cli/src/start/platforms/PlatformManager.ts

```diff
@@ -136,7 +163,22 @@ export class PlatformManager<
       resolveSettings.device
     );
     if (options.runtime === 'expo') {
-      return this.openProjectInExpoGoAsync(resolveSettings);
+      const deviceManager = await this.props.resolveDeviceAsync(resolveSettings);
+      if (this.props.getInterstitialPageUrl()) {
+        const applicationId = await this._getAppIdResolver().getAppIdAsync();
+        if (await deviceManager.isAppInstalledAsync(applicationId)) {
+          return this.openProjectInterstitialPageAsync(deviceManager);
+        } else {
+          Log.warn(
+            `\u203A The 'expo-dev-client' package is installed, but a development build isn't ` +
+              `available.\nYour app will open in Expo Go instead. If you want to use a ` +
+              `development build, you need to make and install one first.\n${learnMore(
+                'https://docs.expo.dev/development/build/'
+              )}`
+          );
```

Comment:
I imagine this would go here: [assertion](https://github.com/expo/expo/blob/9a148a31a58007f230f4f6b2369811c2649bf989/packages/%40expo/cli/src/start/platforms/PlatformManager.ts#L57)?

Reply:
Since this log message is specific to the case where no dev build is installed on the device, but the interstitial page _would_ otherwise be opened, it has to live in `shouldUseInterstitialPage` now that the logic has moved there too.